Application/EdFi.Ods.Tests/EdFi.Ods.Features/UniqueIdIntegration/PersonIdentifiersProviderTests.cs

```diff
@@ -0,0 +1,133 @@
+using EdFi.Ods.Api.IdentityValueMappers;
+using EdFi.Ods.Common.Specifications;
+using EdFi.Ods.Common.Extensions;
+using EdFi.TestFixture;
+using FakeItEasy;
+using NUnit.Framework;
+using Shouldly;
+using Test.Common;
+
+namespace EdFi.Ods.Tests.EdFi.Ods.Features.UniqueIdIntegration
+{
+    [SuppressMessage("ReSharper", "InconsistentNaming")]
+    public class When_retrieving_all_the_Student_identity_value_mappings
+        : TestFixtureAsyncBase
+    {
+        // Supplied values
+        private List<PersonIdentifiersValueMap> _suppliedPersonIdentifiers;
+
+        // Actual values
+        private IEnumerable<PersonIdentifiersValueMap> _actualStudentIdentifiers;
+        private IEnumerable<PersonIdentifiersValueMap> _actualStaffIdentifiers;
+        private IEnumerable<PersonIdentifiersValueMap> _actualParentIdentifiers;
+
+        // External dependencies
+        private IPersonIdentifiersProvider _personIdentifiersProvider;
+
+        protected override async Task ArrangeAsync()
+        {
+            _personIdentifiersProvider = Stub<IPersonIdentifiersProvider>();
+
+            _suppliedPersonIdentifiers = new List<PersonIdentifiersValueMap>
+                {
+                    new PersonIdentifiersValueMap
+                    {
+                        UniqueId = Guid.NewGuid()
+                            .ToString(),
+                        Usi = 100,
+                        Id = new Guid()
+                    },
+                    new PersonIdentifiersValueMap
+                    {
+                        UniqueId = Guid.NewGuid()
+                            .ToString(),
+                        Usi = 101,
+                        Id = new Guid()
+                    }
+                };
+
+            A.CallTo(() => _personIdentifiersProvider.GetAllPersonIdentifiers("Student"))
+                    .Returns(Task.FromResult((IEnumerable<PersonIdentifiersValueMap>)_suppliedPersonIdentifiers));
+
+            A.CallTo(() => _personIdentifiersProvider.GetAllPersonIdentifiers("Staff"))
+                    .Returns(Task.FromResult((IEnumerable<PersonIdentifiersValueMap>)_suppliedPersonIdentifiers));
+
+            A.CallTo(() => _personIdentifiersProvider.GetAllPersonIdentifiers("Parent"))
+                    .Returns(Task.FromResult((IEnumerable<PersonIdentifiersValueMap>)_suppliedPersonIdentifiers));
+
+            string validPersonTypes = string.Join("','", PersonEntitySpecification.ValidPersonTypes)
+                    .SingleQuoted();
+            A.CallTo(() => _personIdentifiersProvider.GetAllPersonIdentifiers("NonPersonType"))
+                .ThrowsAsync(new AggregateException(new ArgumentException($"Invalid person type 'NonPersonType'. Valid person types are: {validPersonTypes}")));
+        }
+
+        protected override async Task ActAsync()
+        {
+            _actualStudentIdentifiers = await Task.FromResult(_personIdentifiersProvider.GetAllPersonIdentifiers("Student").Result);
```

Comment:
this is still wrong

Reply:
it should be `await _personsIdentifiersProvider.GetAllPrssionIdentifiers("Student:);`

Reply:
updated